src/issue.rs

```diff
@@ -0,0 +1,75 @@
+use crate::db::*;
+use crate::{
+	error,
+	github,
+	github_bot::GithubBot,
+	matrix,
+	matrix_bot::MatrixBot,
+	project,
+	Result,
+};
+use itertools::Itertools;
+use rocksdb::DB;
+use snafu::{
+	GenerateBacktrace,
+	ResultExt,
+};
+use std::collections::HashMap;
+use std::time::{
+	Duration,
+	SystemTime,
+};
+
+fn issue_project(issue: &github::Issue, github_bot: &GithubBot) -> Result<Option<github::Project>> {
+	let repo = &issue.repository;
+	let issue_number = error::unwrap_field(issue.number)?;
```

Comment:
As above can be replaced with `use snafu::Backtrace;`, and `issue.number.context(error::MissingData)?;` instead.